File: Core/device/devicediscoverymessage.cpp
#include "devicediscoverymessage.h"
#include "shared/device.h"
#include "shared/mqtt_qt.h"

#include <QDateTime>

DeviceDiscoveryMessage::DeviceDiscoveryMessage(QString deviceId, QString serviceId, quint64 upTime, DeviceHealthState healthState, QObject *parent) : MessageBase(parent), m_deviceId(deviceId), m_serviceId(serviceId), m_upTime(upTime), m_healthState(healthState)
{
}

QString DeviceDiscoveryMessage::fullId() {
    return m_deviceId + DEVICE_FULLID_SEP + m_serviceId;
}

QString DeviceDiscoveryMessage::deviceId() {
    return m_deviceId;
}

QString DeviceDiscoveryMessage::serviceId() {
    return m_serviceId;
}

MessageBase::MESSAGE_TYPE DeviceDiscoveryMessage::getMessageType() {
    return MESSAGE_TYPE_DEVICE_DISCOVERY;
}

QString DeviceDiscoveryMessage::getFirstLevelId() {
    return m_deviceId;
}

QString DeviceDiscoveryMessage::getSecondLevelId() {
    return m_serviceId;
}

quint64 DeviceDiscoveryMessage::upTime() {
    return m_upTime;
}

DeviceDiscoveryMessage::DeviceHealthState DeviceDiscoveryMessage::healthState() {
    return m_healthState;
}

File: AndroidApp/OshCore/src/main/java/com/osh/device/DeviceDiscoveryMessage.java
package com.osh.device;

import com.osh.communication.MessageBase;

public class DeviceDiscoveryMessage extends MessageBase {
	
    private String deviceId;
    private String serviceId;

	public DeviceDiscoveryMessage(String deviceId, String serviceId) {
		this.deviceId = deviceId;
		this.serviceId = serviceId;
	}

	@Override
	public MESSAGE_TYPE getMessageType() {
		return MESSAGE_TYPE.MESSAGE_TYPE_DEVICE_DISCOVERY;
	}

	@Override
	public String getFirstLevelId() {
		return deviceId;
	}

	@Override
	public String getSecondLevelId() {
		return serviceId;
	}

	public String getDeviceId() {
		return deviceId;
	}

	public void setDeviceId(String deviceId) {
		this.deviceId = deviceId;
	}

	public String getServiceId() {
		return serviceId;
	}

	public void setServiceId(String serviceId) {
		this.serviceId = serviceId;
	}
	
}
